models/product.go

```diff
@@ -0,0 +1,53 @@
+package models
+
+import (
+	"github.com/google/uuid"
+)
+
+type Product struct {
+	ID      uuid.UUID `validation:"required"`
+	Name    string    `validation:"required"`
+	Public  bool      `validation:"required"`
```

Comment:
Simple bool couldn't be omited or required.
If you really would like to make bool values required, you should use pointers here:
```suggestion
	Public  *bool      `validation:"required"`
```